src/Freya.Integration/Integration.fs

```diff
@@ -0,0 +1,164 @@
+    /// to allow use of standard OWIN components within Freya.
+    /// NOTE: EXPERIMENTAL
+    [<CompiledName ("ToFreya")>]
+    let toFreya (app: OwinAppFunc) : Freya<unit> =
+        fun s -> async {
+            let! token = Async.CancellationToken
+            // Apply and mutate the OwinEnvironment asynchronously
+            do! Async.AwaitTask <| app.Invoke(s.Environment).ContinueWith<unit>((fun _ -> ()), token)
+            // Return the result as a unit value and the mutated FreyaState
+            return (), s }
+
+open Freya.Pipeline
+
+/// Type alias for the OWIN MidFunc signature.
+type OwinMidFunc =
+    Func<OwinAppFunc, OwinAppFunc>
+
+/// Provides transformation functions for converting to/from OWIN middlewares from/to Freya Pipelines.
+[<RequireQualifiedAccess>]
+[<CompilationRepresentation (CompilationRepresentationFlags.ModuleSuffix)>]
+module OwinMidFunc =
+    
+    /// Converts a Freya.Pipeline to an OWIN MidFunc run before executing the next OwinAppFunc.
+    [<CompiledName("FromFreya")>]
+    let ofFreya (pipeline: Freya<FreyaPipelineChoice>) =
+        OwinMidFunc(fun next ->
+            let app e =
+                // Convert to FreyaState
+                let s = { Environment = e
+                          Meta = { Memos = Map.empty } }
+                async {
+                    // Execute the pipeline
+                    let! choice, s' = pipeline s
+                    match choice with
+                    // Execute the next OwinAppFunc
+                    | Next ->
+                        return! next.Invoke(s'.Environment).ContinueWith<unit>(fun _ -> ()) |> Async.AwaitTask
+                    // Complete the Task
+                    | Halt -> return () }
+                |> Async.StartAsTask :> Task
+            OwinAppFunc app)
+    
+    /// Converts a Freya.Pipeline to an OWIN MidFunc run after executing the next OwinAppFunc.
+    [<CompiledName("FromFreyaAfter")>]
+    let ofFreyaAfter (pipeline: Freya<FreyaPipelineChoice>) =
+        OwinMidFunc(fun next ->
+            let app e =
+                async {
+                    let task = next.Invoke e
+                    do! task.ContinueWith<unit>(fun _ -> ()) |> Async.AwaitTask
+                    // Check the result. If the Task is complete run the after pipeline.
+                    if (task.Status = TaskStatus.RanToCompletion) then
+                        // Convert to FreyaState
+                        let s = { Environment = e
+                                  Meta = { Memos = Map.empty } }
+                        // Execute the pipeline
+                        do! pipeline s |> Async.Ignore
+                    else return () }
+                |> Async.StartAsTask :> Task
+            OwinAppFunc app)
+
+    /// Converts a Freya.Pipeline to an OWIN MidFunc.
+    [<CompiledName("FromFreyaWrapped")>]
+    let ofFreyaWrapped (before: Freya<FreyaPipelineChoice>) (after: Freya<FreyaPipelineChoice>) =
```

Comment:
I really don't like this name. Also, I've been thinking it would be nice to have the opposite set of methods as extensions on `Freya`.